R/utility.R

```diff
@@ -20,13 +20,13 @@ as_cordf <- function(x, diagonal = NA) {
   }
   x <- as.data.frame(x)
   row_name <- x$rowname
-  x <- x[, colnames(x) != "rowname"]
+  x <- x[colnames(x) != "rowname"]
   rownames(x) <- row_name
   if(ncol(x) != nrow(x)) {
     stop("Input object x is not a square. ",
          "The number of columns must be equal to the number of rows.")
   }
-  diag(x) <- diagonal
+  if (ncol(x) > 1) diag(x) <- diagonal
```

Comment:
These changes fix the output when `x` and `y` are numeric vectors. This would be an example of a use case that goes away if the `y` argument is removed. (See previous discussion in #115).

Reply:
It's not _only_ there to fix the `x`,`y` use case. It also ensures we get a meaningful result with a one-column data.frame.

```
# Correlation of a df with only one column
mtcars %>% 
  select(cyl) %>% 
  correlate(quiet = TRUE)
#> # A tibble: 1 x 2
#>   rowname   cyl
#>   <chr>   <dbl>
#> 1 cyl         1
```